### PR TITLE
Implement exporting to Gltf

### LINKED ISSUE #264: Export to glTF
Initially suggested in https://github.com/CadQuery/cadquery/issues/105.

JSON (I think that is what is known as `TJS` in CadQuery) is deprecated in Three.js, and other formats seem to be lossy formats or 2D formats (like SVG).

As @fragmuffin mentioned:

> `cqparts` also has a [GLTF exporter](https://cqparts.github.io/cqparts/doc/api/cqparts.codec.html#module-cqparts.codec.gltf), which is just awesome (GLTF I mean :wink:)

Maybe some code can be reused.

### PR DESCRIPTION
**Pull request by adam-urbanczyk**
Will resolve #264

### CODE REVIEW

**Review by marcus7070 — COMMENTED** [review]
When I save `nested_assy` to GLTF through the interpreter I get
```python
>>> assy.save("nested.glb", "GLTF")
RWGltf_CafWriter skipped node 'TOP_part' without triangulation data
RWGltf_CafWriter skipped node 'SECOND_part' without triangulation data
RWGltf_CafWriter skipped node 'BOTTOM_part' without triangulation data
<cadquery.assembly.Assembly object at 0x7f0fb07f0340>
>>> 
```

and a `nested.glb` file that has no data in it.

Commenting out the earlier saves in the test also produces the same `nested.glb`, I think one of the earlier saves is triangulating it.

**Comment by adam-urbanczyk** [comment]
Thanks, missed that.

**Comment by jmwright** [comment]
GitHub is telling me that I'm the only reviewer holding this up, but I still get the same error as @marcus7070 . Are there changes yet to come on this branch?

**Comment by adam-urbanczyk** [comment]
It still needs to be fixed.

**Comment by adam-urbanczyk** [comment]
As indicated above it needs to be fixed.

**Comment by adam-urbanczyk** [comment]
I have no understanding of GLTF myself, this feature will use what is provided by OCCT and export an assy (including colors).

**Comment by adam-urbanczyk** [comment]
Should be fine now @marcus7070 @jmwright .

**Review by jmwright — APPROVED** [review]
Thanks for putting this together @adam-urbanczyk 

Looks good to me.

File: tests/test_assembly.py [hunk 1]
```diff
@@ -168,28 +184,41 @@ def test_toJSON(simple_assy, nested_assy, empty_top_assy):
...
+    assert os.path.exists("nested.xml")
+
+    nested_assy.save("nested.step")
+    assert os.path.exists("nested.step")
+
+    nested_assy.save("nested.stp", "STEP")
+    assert os.path.exists("nested.stp")
+
+    nested_assy.save("nested.caf", "XML")
+    assert os.path.exists("nested.caf")
```
Comment (marcus7070) [inline tests/test_assembly.py:202]: Are you interested in converting this to a pytest parameterized test? If so just let me know and I'll push the following (GitHub won't let me make suggestions over deleted lines)
```python
@pytest.mark.parametrize(
    "extension, args",
    [
        ("step", ()),
        ("xml", ()),
        ("stp", ("STEP",)),
        ("caf", ("XML",)),
        ("wrl", ("VRML",)),
    ],
)
def test_save(extension, args, nested_assy, nested_assy_sphere):

    filename = "nested." + extension
    nested_assy.save(filename, *args)
    assert os.path.exists(filename)


def test_save_gltf(nested_assy_sphere):

    nested_assy_sphere.save("nested.glb", "GLTF")
    assert os.path.exists("nested.glb")
    assert os.path.getsize("nested.glb") > 50 * 1024


def test_save_vtkjs(nested_assy):

    nested_assy.save("nested", "VTKJS")
    assert os.path.exists("nested.zip")
```

Parameterize would prevent problem from a previous comment, where the first save changed the state of the assembly and affected testing a following save.

If you're not interested note that `nested_assy.save("nested.step")` currently appears twice.

Comment (adam-urbanczyk) [reply tests/test_assembly.py:202]: Good idea, please do push!

**Review by marcus7070 — APPROVED** [review]

**Pull request merged**